x-pack/plugins/lens/public/indexpattern_datasource/dimension_panel/filtering.tsx

```diff
@@ -49,22 +60,40 @@ export function Filtering({
   updateLayer: (newLayer: IndexPatternLayer) => void;
   isInitiallyOpen: boolean;
 }) {
+  const inputFilter = selectedColumn.filter;
+  const [queryInput, setQueryInput] = useState(inputFilter ?? defaultFilter);
   const [filterPopoverOpen, setFilterPopoverOpen] = useState(isInitiallyOpen);
+
+  const onQueryInputChange = useCallback(
+    (newQuery) => {
+      setQueryInput(newQuery);
+    },
+    [setQueryInput]
+  );
```

Comment:
It looks there's no difference here with `setQueryInput` itself. Maybe we can remove this?